src/filter.js

```diff
@@ -111,6 +112,10 @@ Filter.convertToRegExpString = function(regex) {
     return regex.toString();
   }
 
+  if (Buffer.isBuffer(regex)) {
+    return escapeStringRegexp(regex.toString());
```

Comment:
This will create an utf-8 string. Using `.toString('binary')` will preserve the actual bytes

Reply:
```js
Buffer.from(Buffer.from('e6','hex').toString())
// <Buffer ef bf bd>

Buffer.from(Buffer.from('e6','hex').toString('binary'))
// <Buffer c3 a6>

Buffer.from(Buffer.from('e6','hex').toString('binary'), 'binary')
// <Buffer e6>
```

Reply:
Maybe use `\xFF` escaping for non-ASCII bytes?